File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n=len(nums)
        array=[0]*n
        low=0
        high=n-1
        n=n-1
        while(low<=high):
            if(abs(nums[low])>abs(nums[high])):
                array[n]=nums[low]*nums[low]
                low=low+1
            else:
                array[n]=nums[high]*nums[high]
                high=high-1
            n=n-1
            
        return array

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let array=new Array(nums.length);
    let low=0,high=nums.length-1;
    let ind=high;
    while(low<=high)
    {
        
        if(Math.abs(nums[low])>Math.abs(nums[high]))
        {
            array[ind]=(nums[low]*nums[low]);
            low++;
        }
        else
        {
            array[ind]=(nums[high]*nums[high]);
            high--;
        }
        ind--;
    }
    return array;
};